src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -223,6 +225,19 @@ void CsvImportWidget::writeDatabase() {
         entry->setPassword(m_parserModel->data(m_parserModel->index(r, 3)).toString());
         entry->setUrl(m_parserModel->data(m_parserModel->index(r, 4)).toString());
         entry->setNotes(m_parserModel->data(m_parserModel->index(r, 5)).toString());
+
+        if (m_parserModel->data(m_parserModel->index(r, 6)).isValid()) {
+            qint64 lastModified = m_parserModel->data(m_parserModel->index(r, 6)).toString().toLongLong();
+            qint64 created = m_parserModel->data(m_parserModel->index(r, 7)).toString().toLongLong();
```

Comment:
You are assuming that index 7 is valid based on the status of index 6. This is not correct and could result in undefined behavior. Recommend splitting index 7 into it's own validated code block.

Reply:
I think that's a good idea. Does the code block starting at [Line 218](https://github.com/keepassxreboot/keepassxc/pull/1359/files/ef974e101b9ce271fba9e211d9d6ae711d60cb4d#diff-4197800e8a547e66805cc5e10f23b629R218) have the same problem? If so, should we an open issue to validate each index separately?

Reply:
Well, the approach is explained in the comment at line 217... But that's completely another kind of problem: here fields are completely unrelated, which doesn't hold for creation and mod dates.